lib/fast_gettext/translation.rb

```diff
@@ -128,44 +122,98 @@ def ds_(domain, key, separator=nil, &block)
       end
     end
 
+    def dp_(domain, namespace, key, separator=nil, &block)
+      _in_domain domain do
+        FastGettext::Translation.p_(namespace, key, separator, &block)
+      end
+    end
+
     def dns_(domain, *keys, &block)
       _in_domain domain do
         FastGettext::Translation.ns_(*keys, &block)
       end
     end
 
+    def dnp_(domain, context, key, *args)
+      _in_domain domain do
+        result = FastGettext::Translation.np_(context, key, *args){nil}
```

Comment:
should pass the block here instead of extra yield logic below ?

Reply:
Trying to avoid passing block.